crates/server/src/admin/service.rs

```diff
@@ -174,24 +178,96 @@ pub(crate) fn setup(
     Some((admin_path, admin_router))
 }
 
-pub(crate) fn site(
-    config: &ServerConfig,
-) -> Option<(&'static str, ServeDir<SetStatus<ServeFile>>)> {
+/// Creates a router for serving static node-ui files and providing fallback to `index.html` for SPA routing.
+///
+/// This function checks if the admin dashboard is enabled in the provided configuration.
+/// If the admin site is enabled, it returns a router that serves embedded static files
+/// and routes all SPA-related requests (like `/admin-dashboard/`) to `index.html`.
+///
+/// # Parameters
+/// - `config`: A reference to the server configuration that contains the admin site settings.
+///
+/// # Returns
+/// - `Option<(&'static str, Router)>`: If the admin site is enabled, it returns a tuple containing
+///   the base path ("/admin-dashboard") and the router for that path. If the admin site is disabled,
+///   it returns `None`.
+pub(crate) fn site(config: &ServerConfig) -> Option<(&'static str, Router)> {
     let _config = match &config.admin {
         Some(config) if config.enabled => config,
         _ => {
             info!("Admin site is disabled");
             return None;
         }
     };
+
     let path = "/admin-dashboard";
 
-    let react_static_files_path = "./node-ui/build";
-    let react_app_serve_dir = ServeDir::new(react_static_files_path).not_found_service(
-        ServeFile::new(format!("{react_static_files_path}/index.html")),
-    );
+    // Create a router to serve static files and fallback to index.html
+    let router = Router::new()
+        .route("/", get(serve_embedded_file)) // Match /admin-dashboard
+        .route("/*path", get(serve_embedded_file)); // Match /admin-dashboard/* for all sub-paths
+
+    Some((path, router))
+}
+
+/// Serves embedded static files or falls back to `index.html` for SPA routing.
+///
+/// This function handles requests by removing the "/admin-dashboard/" prefix from the requested URI path,
+/// and then attempting to serve the requested file from the embedded directory. If the requested file
+/// is not found, it serves `index.html` to support client-side routing.
+///
+/// # Parameters
+/// - `uri`: The requested URI, which will be used to determine the file path in the embedded directory.
+///
+/// # Returns
+/// - `Result<impl IntoResponse, StatusCode>`: If the requested file is found or the fallback to index.html
+///   succeeds, it returns an `Ok` with the response. If no file can be served, it returns an `Err` with
+///   a 404 NOT_FOUND status code.
+async fn serve_embedded_file(uri: Uri) -> Result<impl IntoResponse, StatusCode> {
+    // Extract the path from the URI, removing the "/admin-dashboard/" prefix and any leading slashes
+    let path = uri
+        .path()
+        .trim_start_matches("/admin-dashboard/")
+        .trim_start_matches('/')
+        .to_string();
```

Comment:
seems this can be skipped?

```suggestion
        .trim_start_matches('/');
```